### PR TITLE
planner, executor: support create view on union

### PR DESCRIPTION
**Pull request by XuHuaiyu**
related PR:
https://github.com/pingcap/parser/pull/572

### What problem does this PR solve? 
1. support create view on union
2. refine the error message to make it more compatible with MySQL

### What is changed and how it works?
1. change go.mod
2. reset the error to `ErrViewInvalid` when buildViewFromDatasource raises an error during building the logical plan

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported function/method change

Side effects

N/A

Related changes

 - Need to cherry-pick to the release branch

Release note

[bug-fix] Support create view on a union statement. Creating view on a union statement will cause a syntax error before.

### CODE REVIEW

File: executor/ddl_test.go [hunk 1]
```diff
@@ -224,6 +225,33 @@ func (s *testSuite3) TestCreateView(c *C) {
...
+	tk.MustExec("create table t2(a int, b int)")
+	tk.MustExec("insert into t1 values(1,2), (1,1), (1,2)")
+	tk.MustExec("insert into t2 values(1,1),(1,3)")
+	tk.MustExec("create definer='root'@'localhost' view v as select * from t1 union select * from t2")
+	tk.MustQuery("select * from v").Sort().Check(testkit.Rows("1 1", "1 2", "1 3"))
+	tk.MustExec("alter table t1 drop column a")
+	_, err = tk.Exec("select * from v")
+	c.Assert(terror.ErrorEqual(err, plannercore.ErrViewInvalid), IsTrue)
+	tk.MustExec("alter table t1 add column a int")
+	tk.MustQuery("select * from v").Sort().Check(testkit.Rows("1 1", "3 1", "<nil> 1", "<nil> 2"))
```
Comment (wshwsh12) [inline executor/ddl_test.go]: I think the test case is wrong.  
It should be `tk.MustQuery("select * from v").Sort().Check(testkit.Rows("1 1", "1 3", "<nil> 1", "<nil> 2"))`
The result is different from MySQL.

```
mysql> select * from v;
+------+------+
| a    | b    |
+------+------+
| NULL |    2 |
| NULL |    1 |
|    1 |    1 |
|    1 |    3 |
+------+------+
4 rows in set (0.00 sec)

tidb> select * from v;
+---+---+
| a | b |
+---+---+
| NULL | 2 |
| 3 | 1 |
| NULL | 1 |
| 1 | 1 |
+---+---+
4 rows in set (0.00 sec)
```

**Comment by XuHuaiyu** [comment]
@SunRunAway CI failed before I run `go mod tidy`.

**Pull request merged**